e2e/e2e_test.go

```diff
@@ -0,0 +1,135 @@
+
+			selector := labels.NewSelector().Add(*req)
+
+			workers := &v1.NodeList{}
+			Expect(k8sClient.List(context.Background(), workers, &client.ListOptions{LabelSelector: selector})).ToNot(HaveOccurred())
+			Expect(len(workers.Items)).To(BeNumerically(">=", 2))
+			node = &workers.Items[0]
+			machine = getAssociatedMachine(node)
+		})
+
+		AfterEach(func() {
+			if mdr != nil {
+				deleteRemediation(mdr)
+			}
+		})
+
+		Describe("CR created for an unhealthy node", func() {
+			It("recreates the associated Machine", func() {
+				mdr = createRemediation(node)
+
+				By("checking the Machine was deleted")
+				Eventually(func() bool {
+					key := client.ObjectKeyFromObject(machine)
+					err := k8sClient.Get(context.TODO(), key, getMachine())
+					return errors.IsNotFound(err)
+				}, 5*time.Minute, 10*time.Second).Should(BeTrue())
+
+				By("checking the Node is recreated")
+				newNode := &v1.Node{}
+				Eventually(func() types.UID {
+					key := client.ObjectKeyFromObject(node)
+					if err := k8sClient.Get(context.TODO(), key, newNode); err != nil {
+						return node.GetUID()
+					}
+					newUID := newNode.GetUID()
+					return newUID
+				}, 15*time.Minute, 10*time.Second).ShouldNot(Equal(node.GetUID()))
+
+				By("checking associated Machine was created after the remediation")
+				newMachine := getAssociatedMachine(newNode)
+				Expect(newMachine.GetUID()).ShouldNot(Equal(machine.GetUID()))
+				Expect(newMachine.GetCreationTimestamp().Time).Should(BeTemporally(">=", mdr.GetCreationTimestamp().Time))
+			})
+		})
+	})
+})
+
+func getMachine() *unstructured.Unstructured {
+	machine := new(unstructured.Unstructured)
+	machine.SetKind(machineKind)
+	machine.SetAPIVersion(machineAPIVersion)
+	return machine
+
+}
+
+func getAssociatedMachine(node *v1.Node) *unstructured.Unstructured {
+	machineNamespace := strings.Split(node.GetAnnotations()[machineAnnotationOpenshift], "/")[0]
+	machineName := strings.Split(node.GetAnnotations()[machineAnnotationOpenshift], "/")[1]
+	key := client.ObjectKey{
+		Name:      machineName,
+		Namespace: machineNamespace,
+	}
+
+	machine := getMachine()
+	Expect(k8sClient.Get(context.TODO(), key, machine)).To(Succeed())
+	return machine
+}
+
+func createRemediation(node *v1.Node) *v1alpha1.MachineDeletion {
+	mdr := &v1alpha1.MachineDeletion{}
+	mdr.Name = node.Name
+	mdr.Namespace = machineDeletionNamespace
+
+	ExpectWithOffset(1, k8sClient.Create(context.Background(), mdr)).ToNot(HaveOccurred())
+	return mdr
+}
+
+func deleteRemediation(mdr *v1alpha1.MachineDeletion) {
+	timeout := 2 * time.Minute
+	pollInterval := 10 * time.Second
+	// Delete
+	EventuallyWithOffset(1, func() error {
+		err := k8sClient.Delete(context.Background(), mdr)
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}, timeout, pollInterval).ShouldNot(HaveOccurred(), "failed to delete mdr")
+
+	// Wait until deleted
+	EventuallyWithOffset(1, func() bool {
```

Comment:
Why do we need this ?
IIUC in the previous statement we already verify that the MDR doesn't exist

Reply:
true, I will remove it